api/spec/requests/spree/api/v2/storefront/products_spec.rb

```diff
@@ -613,6 +613,150 @@
+
+      let!(:unused_option_type) { create(:option_type) }
+      let!(:unused_option_value) { create(:option_value) }
+
+      let(:option_type1_response) do
+        {
+          id: option_type.id,
+          name: option_type.name,
+          presentation: option_type.presentation,
+          option_values: [
+            {
+              id: option_value.id,
+              name: option_value.name,
+              presentation: option_value.presentation,
+              position: option_value.position
+            }
+          ]
+        }
+      end
+
+      let(:option_type2_response) do
+        {
+          id: option_type2.id,
+          name: option_type2.name,
+          presentation: option_type2.presentation,
+          option_values: [
+            {
+              id: option_type2_value1.id,
+              name: option_type2_value1.name,
+              presentation: option_type2_value1.presentation,
+              position: option_type2_value1.position
+            },
+            {
+              id: option_type2_value2.id,
+              name: option_type2_value2.name,
+              presentation: option_type2_value2.presentation,
+              position: option_type2_value2.position
+            }
+          ]
+        }
+      end
+
+      let(:property2_response) do
+        {
+          id: property2.id,
+          name: property2.name,
+          presentation: property2.presentation,
+          values: [
+            {
+              value: product_property2.value,
+              filter_param: product_property2.filter_param,
+            },
+            {
+              value: product2_property2.value,
+              filter_param: product2_property2.filter_param,
+            }
+          ]
+        }
+      end
+
+      let(:property3_response) do
+        {
+          id: property3.id,
+          name: property3.name,
+          presentation: property3.presentation,
+          values: [
+            {
+              value: product2_property3.value,
+              filter_param: product2_property3.filter_param
+            }
+          ]
+        }
+      end
+
+      context 'when no filters are applied' do
+        before { get '/api/v2/storefront/products' }
+
+        it 'returns list of available filters for all products' do
+          expect(json_response['meta']['filters']['option_types'].count).to eq 2
+          expect(json_response['meta']['filters']['option_types']).to include(option_type1_response)
+          expect(json_response['meta']['filters']['option_types']).to include(option_type2_response)
+          expect(json_response['meta']['filters']['product_properties'].count).to eq 2
+          expect(json_response['meta']['filters']['product_properties']).to include(property2_response)
+          expect(json_response['meta']['filters']['product_properties']).to include(property3_response)
+        end
+      end
+
+      context 'when filter by option type is applied' do
```

Comment:
Both contexts for filtering by option types and product properties are returning the same filters as for the context where you're not applying any filters. Could we change that so responses are different each time?

We can also use `to contain_exactly(...)` instead of `to_include(...)` to avoid false negatives.

Reply:
Good point with the contain_exactly.

As for the contexts comment - when an option_type or property filter is applied, it shouldn't change available filters returned in metadata, right? 
My understanding is that only filtering by taxons can change what's returned here. That's also how I designed these test cases - all three scenarios should result in the same output.

Let me know if that makes sense, if I'm wrong I'll review it one more time ;)